### PR TITLE
Low Memory Transformer's PR - [WIP]

### PR DESCRIPTION
**Pull request by johntremblay**
Just opening the PR for potential comments and showing the progress.

It is still a work in progress [WIP] for now.

### CODE REVIEW

**Comment by johntremblay** [comment]
@BassirouNdao As of now, I added the Longformer and Linformer for the LibriSpeech. I suggest we use the current branch as our dev and we can branch out from this one and do PR for ourselves and finally just merge this one into the main SB's repo.

File: .gitignore [hunk 1]
```diff
@@ -143,3 +143,6 @@ dmypy.json
 
 # Mac OS
 .DS_Store
+
+# IntelliJ IDE idea folder
+.idea/
```
Comment (Arka161) [inline .gitignore:148]: I think this diff should be reverted

Comment (johntremblay) [reply .gitignore:148]: @Arka161 I tried to find a solution to exclude the ideas folder from pycharm but wasnt able to find something convincing. Do you have a solution in mind?

Comment (Arka161) [reply .gitignore:148]: @johntremblay I basically meant something like avoiding any new lines in the `.gitignore`, and checking out this file from `develop` as it's just an IDE change, and not essential for speechbrain. Though it's just a clean coding convention.

**Comment by johntremblay** [comment]
Will be tackled in another PR

**Pull request closed without merging**